src/components/WeatherWidget/WeatherWidget.tsx

```diff
@@ -128,30 +99,7 @@ export default function WeatherWidget({ location }: Props) {
                   // We should leave some cells empty because the weather API doesn't provide data for the past hours of the current day. Also, some cells at the end of the table are empty because data is only provided for the next 5 days
                   <td key={idx}> </td>
                 ))}
-              {weather.map((weatherReport) => {
-                const {
-                  dt,
-                  weather: [{ icon, description }],
-                  main: { temp, humidity },
-                } = weatherReport;
-                return (
-                  <td key={dt}>
-                    <Tooltip title={description}>
-                      <img
-                        src={`${
-                          import.meta.env.VITE_BASE_URL
-                        }img/wn/${icon}.png`}
-                        alt='Weather condition'
-                      />
-                    </Tooltip>
-                    {`Temperature: ${formatTemperatureData(
-                      temp,
-                      selectedTemperatureUnit,
-                    )}
-                    Humidity: ${humidity}%`}
-                  </td>
-                );
-              })}
+              <TableCell weatherReport={weather} />
```

Comment:
Just `weather` prop

Reply:
I can't call it weather, because I have 'weather' key in response data from API. It causes a conflict.

Reply:
There is no conflict is this component because you destructure not this key but properties of `weather` key. It means that constant with this name won't be created